mlflow/store/model_registry/file_store.py

```diff
@@ -0,0 +1,767 @@
+        :param source: Source path where the MLflow model is stored.
+        :param run_id: Run ID from MLflow tracking server that generated the model.
+        :param tags: A list of :py:class:`mlflow.entities.model_registry.ModelVersionTag`
+                     instances associated with this model version.
+        :param run_link: Link to the run from an MLflow tracking server that generated this model.
+        :param description: Description of the version.
+        :return: A single object of :py:class:`mlflow.entities.model_registry.ModelVersion`
+                 created in the backend.
+        """
+
+        def next_version(registered_model):
+            if registered_model.latest_versions:
+                return max([mv.version for mv in registered_model.latest_versions]) + 1
+            else:
+                return 1
+
+        _validate_model_name(name)
+        for tag in tags or []:
+            _validate_model_version_tag(tag.key, tag.value)
+        for attempt in range(self.CREATE_MODEL_VERSION_RETRIES):
+            try:
+                creation_time = now()
+                registered_model = self._get_registered_model(name)
+                registered_model.last_updated_timestamp = creation_time
+                version = next_version(registered_model)
+                model_version = ModelVersion(
+                    name=name,
+                    version=version,
+                    creation_timestamp=creation_time,
+                    last_updated_timestamp=creation_time,
+                    description=description,
+                    current_stage=None,
+                    source=source,
+                    run_id=run_id,
+                    run_link=run_link,
+                )
+                model_version.tags = [
+                    ModelVersionTag(key=tag.key, value=tag.value) for tag in tags or []
+                ]
+                model_version_dir = self._get_model_version_dir(name, version)
+                mkdir(model_version_dir)
+                model_version_dict = dict(model_version)
+                del model_version_dict["tags"]
+                write_yaml(model_version_dir, FileStore.META_DATA_FILE_NAME, model_version_dict)
+                if tags is not None:
+                    for tag in tags:
+                        self.set_model_version_tag(name, version, tag)
+                return model_version
+            except Exception:
+                more_retries = self.CREATE_MODEL_VERSION_RETRIES - attempt - 1
+                logging.warning(
+                    "Model Version creation error (name=%s) Retrying %s more time%s.",
+                    name,
+                    str(more_retries),
+                    "s" if more_retries > 1 else "",
+                )
+        raise MlflowException(
+            "Model Version creation error (name={}). Giving up after "
+            "{} attempts.".format(name, self.CREATE_MODEL_VERSION_RETRIES)
+        )
```

Comment:
Can we move this inside the `except Exception` and throw it if we're out of retries? That way we can make this a raise `MlflowException from exc` (where `exc` is the caught `Exception`) so that it's easy to see what the failure was. Right now, there isn't any exception content to debug with.